bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/NewActionProvider.java

```diff
@@ -75,9 +81,22 @@ public void init(ICommonActionExtensionSite anExtensionSite) {
 			showDlgAction = ActionFactory.NEW.create(window);
 			newProjectAction = new NewProjectAction(window);
 			newExampleAction = new NewExampleAction(window);
-
+			Predicate<IWizardDescriptor> projectWizardFilter = wizardDescriptor -> {
+				String[] tags = wizardDescriptor.getTags();
+				for (String tag : tags) {
+					if (WorkbenchWizardElement.TAG_PROJECT.equals(tag)) {
+						return true;
+					}
+				}
+				return false;
+			};
```

Comment:
Arrays.asList(wizardDescriptor.getTags()).contains(WorkbenchWizardElement.TAG_PROJECT)

Reply:
I'm not sure if this is any better here beside that it creates a new collection?!?

Reply:
It's definitely easier to read and it tells explicitly in 1 line the intent that 6 lines of code doesn't express as clearly (expressing more about the documentation than about the goal).

Reply:
done...